src/language/schemaParser.js

```diff
@@ -0,0 +1,246 @@
+/* @flow */
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import { Source } from './source';
+
+import {
+  ParseOptions,
+  ParseOps,
+} from './parser';
+
+import { TokenKind } from './lexer';
+
+import type {
+  Document,
+  TypeDefinition,
+  FieldDefinition,
+  EnumDefinition,
+  InterfaceDefinition,
+  ArgumentDefinition,
+  ScalarDefinition,
+  InputObjectDefinition,
+} from './ast';
+
+import {
+  DOCUMENT,
+  ENUM_DEFINITION,
+  TYPE_DEFINITION,
+  INTERFACE_DEFINITION,
+  FIELD_DEFINITION,
+  ARGUMENT_DEFINITION,
+  UNION_DEFINITION,
+  SCALAR_DEFINITION,
+  INPUT_OBJECT_DEFINITION,
+} from './kinds';
+
+export function parseSchema(
+  source: Source | string,
+  options?: ParseOptions
+): Document {
+  var sourceObj = source instanceof Source ? source : new Source(source);
+  var parser = ParseOps.makeParser(sourceObj, options || {});
+  return parseSchemaDocument(parser);
+}
+
+function parseNextSchemaDefinition(parser): any {
+  switch (parser.token.value) {
+    case 'type':
+      return parseTypeDefinition(parser);
+    case 'enum':
+      return parseEnumDefinition(parser);
+    case 'interface':
+      return parseInterfaceDefinition(parser);
+    case 'union':
+      return parseUnionDefinition(parser);
+    case 'input':
+      return parseInputObjectDefinition(parser);
+    case 'scalar':
+      return parseScalarDefinition(parser);
+    default:
+      throw ParseOps.unexpected(parser);
+  }
+}
+
+function parseScalarDefinition(parser): ScalarDefinition {
+  var start = parser.token.start;
+  ParseOps.expectKeyword(parser, 'scalar');
+  var name = ParseOps.parseName(parser);
+  var loc = ParseOps.loc(parser, start);
+  return {
+    kind: SCALAR_DEFINITION,
+    name: name,
+    loc: loc,
+  };
+}
+
+function parseSchemaDocument(parser) {
+  var start = parser.token.start;
+  var definitions = [];
+  do {
+    if (!ParseOps.peek(parser, TokenKind.NAME)) {
+      throw ParseOps.unexpected(parser);
+    }
+    definitions.push(parseNextSchemaDefinition(parser));
+  } while (!ParseOps.skip(parser, TokenKind.EOF));
+
+  return {
+    kind: DOCUMENT,
```

Comment:
probably want to define this something other than DOCUMENT so the AST isn't overloaded.